pydantic_extra_types/coordinate.py

```diff
@@ -12,116 +12,86 @@ class Latitude(float):
-        elif isinstance(value, str):
-            self._latitude, self._longitude = parse_str(value)
-        elif isinstance(value, Coordinate):
-            self._latitude = value.latitude
-            self._longitude = value.longitude
-        else:
-            raise PydanticCustomError(
-                'coordinate_error',
-                'value is not a valid Coordinate: value must be a tuple, list or string',
-            )
+    latitude: Latitude
+    longitude: Longitude
 
     @classmethod
     def __get_pydantic_json_schema__(
         cls, core_schema: core_schema.CoreSchema, handler: GetJsonSchemaHandler
     ) -> JsonSchemaValue:
-        field_schema: Dict[str, Any] = {}
-        field_schema.update(type='string', format='coordinate')
+        field_schema: Dict[str, Any] = handler(core_schema)
+        field_schema.update(format='coordinate')
         return field_schema
 
     @classmethod
     def __get_pydantic_core_schema__(
         cls, source: Type[Any], handler: Callable[[Any], CoreSchema]
     ) -> core_schema.CoreSchema:
-        return core_schema.general_plain_validator_function(
-            cls._validate, serialization=core_schema.to_string_ser_schema()
+        return core_schema.general_before_validator_function(
+            cls._validate, handler(source), serialization=core_schema.to_string_ser_schema()
         )
 
     @classmethod
-    def _validate(cls, __input_value: Any, _: Any) -> 'Coordinate':
-        return cls(__input_value)
+    def _validate(cls, value: Any, _: Any) -> Dict[str, Any]:
+        if isinstance(value, ArgsKwargs):
+            _validated_tuple = cls._validate_as_tuple(*value.args, **(value.kwargs or {}))
+        else:
+            _validated_tuple = cls._validate_as_tuple(value)
+        return {'latitude': _validated_tuple[0], 'longitude': _validated_tuple[1]}
 
-    @property
-    def latitude(self) -> 'Latitude':
-        """
-        Get the latitude value of the coordinate.
-        """
-        return self._latitude
+    @classmethod
+    def _validate_as_tuple(cls, *args: Any, **kwargs: Any) -> CoordinateTuple | Tuple[float, float]:
+        if kwargs:
+            if kwargs.keys() != {'latitude', 'longitude'}:
+                raise PydanticCustomError(
+                    'coordinate_error', 'Coordinate constructor accepts only "latitude" and "longitude" kwargs'
+                )
+            return kwargs['latitude'], kwargs['longitude']
+
+        if not args:
+            return 0.0, 0.0
+
+        if len(args) > 2:
+            raise PydanticCustomError(
+                'coordinate_error', "Coordinate doesn't accept more than two positional arguments"
+            )
 
-    @property
-    def longitude(self) -> 'Longitude':
-        """
-        Get the longitude value of the coordinate.
-        """
-        return self._longitude
+        if len(args) == 2:
+            arg = (args[0], args[1])
+        else:
```

Comment:
For the coordinates, do we want to authorize just the latitude?

Reply:
Sorry, I didn't understand the question.